### ISSUE TITLE
Heroku Precompile Fails

### ISSUE THREAD

**Issue by JoeGaebel** [mudbugmedia/critical-path-css-rails#26]
See related issue [heroku-buildpack-google-chrome/issues/38](https://github.com/heroku/heroku-buildpack-google-chrome/issues/38)
I'm running an app on a heroku-16 and cedar-14 stack, and it appears to barf with:

```
remote:        Running: rake assets:precompile
remote:        I, [2018-01-30T03:22:12.590734 #1041]  INFO -- : Writing /tmp/build_ec40d31506f51774e1843471db4e3a6e/public/assets/application-0fdddf6eb42efbc276a22bacd4ef1e878aa7a2f431a9558eabe4c3da0dad2034.js
remote:        I, [2018-01-30T03:22:12.591212 #1041]  INFO -- : Writing /tmp/build_ec40d31506f51774e1843471db4e3a6e/public/assets/application-0fdddf6eb42efbc276a22bacd4ef1e878aa7a2f431a9558eabe4c3da0dad2034.js.gz
remote:        (node:1058) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Failed to launch chrome!
remote:        /tmp/build_ec40d31506f51774e1843471db4e3a6e/vendor/bundle/ruby/2.5.0/gems/critical-path-css-rails-2.3.0/node_modules/puppeteer/.local-chromium/linux-508693/chrome-linux/chrome: error while loading shared libraries: libX11-xcb.so.1: cannot open shared object file: No such file or directory
remote:        TROUBLESHOOTING: https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md
remote:        (node:1058) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
remote:        rake aborted!
remote:        Failed to get CSS for route /
```

I've tried adding these buildpacks with no success.
- https://github.com/heroku/heroku-buildpack-google-chrome
- https://github.com/heroku/heroku-buildpack-xvfb-google-chrome
- https://github.com/jontewks/puppeteer-heroku-buildpack
- https://github.com/CoffeeAndCode/puppeteer-heroku-buildpack

**Comment by JoeGaebel** [comment]
Looks like the order of operations here was the problem. Heroku appears to precompile the assets, and then install buildpacks. Because the necessary buildpack wasn't installed, the precompile step would blow up and cancel the deploy. 

In order to move forward, I disabled the critical path precompile step by commenting out the line in`critical_path_css.rake`:
`Rake::Task['assets:precompile'].enhance { Rake::Task['critical_path_css:generate'].invoke }`

Then, I did a deploy, which allowed the buildpack to install. From then on I was able to run `rake critical_path_css:generate`.

Unfortunately I'm still not able to uncomment the rake task, due to the library not being available in the precompilation step.
